tests/console/yast2_settings.pm

```diff
@@ -0,0 +1,85 @@
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: YaST writes config to 70-yast.conf and detects conflicts
+# - Install yast2-network
+# - verify writing config to the correct file
+# - ensure conflict detected when custom confing with high higher priority exist
+# - validate that YaST applies changes to the system. bsc#1167234
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base "y2_module_consoletest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use y2lan_restart_common;
+
+my $yastconf              = '/etc/sysctl.d/70-yast.conf';
+my $customconf_in_etc     = '/etc/sysctl.d/90-custom.conf';
+my $customconf_in_usr_lib = '/usr/lib/sysctl.d/90-custom.conf';
+
+sub verify_write_config {
+    open_yast2_routing();
+    send_key 'spc';
+    assert_screen "ipv4_forwarding_enabled";
+    send_key 'alt-n';
+    wait_serial("yast2-routing-status-0", 180) || die "'yast2-routing' didn't finish";
+    assert_script_run("sysctl net.ipv4.ip_forward | grep 1");
+}
+
+sub ensure_conflict_detected {
+    _create_conflicts();
+    open_yast2_routing();
+    send_key 'spc';
+    assert_screen "ipv4_forwarding_disabled";
+    send_key 'alt-n', wait_screen_change => 1;
+    assert_screen "sysctl_ip_forward_conflict";
+    send_key 'alt-o';
+    wait_serial("yast2-routing-status-0", 180) || die "'yast2-routing' didn't finish";
+}
+
+sub validate_changes_applied_to_system {
+    assert_script_run("cat $yastconf |grep 'net.ipv4.ip_forward = 1'");
+    assert_script_run("sysctl net.ipv4.ip_forward | grep 1");
+    assert_script_run("cat /proc/sys/net/ipv4/ip_forward | grep 1", fail_message => "ip_forward not applied to system bsc#1167234");
+}
+
+sub _create_conflicts {
+    # Files with higher priority that 70-yast.conf.
+    # But /etc has precedence over /usr/lib, thus second one should not
+    # cause conflict
+    my @other_confs = ($customconf_in_etc, $customconf_in_usr_lib);
+    foreach my $conf (@other_confs) {
+        assert_script_run("touch $conf");
+        my $create_other_conf = 'cat > ' . $conf . ' << EOF
+net.ipv4.ip_forward=1
+EOF
+(exit $?)';
+        assert_script_run("$create_other_conf", fail_message => "Creating $conf failed.");
+        script_run("cat $conf", output => "show output for $conf");
+    }
+}
+
+sub run {
+    my $self = shift;
+
+    select_console 'root-console';
+    zypper_call "in yast2-network";    # make sure yast2 routing module installed
+
+    assert_script_run("sysctl net.ipv4.ip_forward | grep 0");
+    verify_write_config;
+    validate_changes_applied_to_system;
+    ensure_conflict_detected;
+    # make sure that nothing has changed
+    validate_changes_applied_to_system;
+    script_run("rm -rf $customconf_in_etc $customconf_in_usr_lib", output => "cleanup test files");
```

Comment:
`assert_script_run` should fit here better.

Reply:
i used script_run instead of the assert_script_run because i dont want to validate the output. I just want to run a cleanup and leave a comment in the logs. I do not need to assert the cmd, as it is not essential part of the code.

Reply:
what happen if `rm` fails due to permissions? I guess next modules should be aware of that, for that reason it is always better to use `assert_script_run` or the test could silently fail even in some not some important step.

Reply:
if it fails you will see a red frame in the test but the test will keep running. and as those file do not affect the rest of the functionality it is quite safe